### PR TITLE
Bugfix/issue 2082 again

### PR DESCRIPTION
**Pull request by bgoodri**
#### Submission Checklist

- [ X] Run unit tests: `./runTests.py src/test/unit`
- [ X] Run cpplint: `make cpplint`
- [ X] Declare copyright holder and open-source license: see below

#### Summary

Add `inline` qualifier to more function

#### Intended Effect

Allow shared objects with multiple Stan programs to be created, such as rstanarm

#### How to Verify

At the moment, run `R CMD SHLIB *.o` with multiple `.o` files that have been compiled by `stan_model`. We are still trying to figure out how to test this in Stan unit tests.

#### Side Effects

Less RAM consumed when building rstanarm

#### Documentation

None

#### Reviewer Suggestions

@syclik 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Trustees of Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/stan/mcmc/hmc/hamiltonians/softabs_point.hpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace stan {
       // Psuedo-Jacobian of the eigenvalues
       Eigen::MatrixXd pseudo_j;
 
-      virtual void
+      virtual inline void
```
Comment (bob-carpenter) [inline src/stan/mcmc/hmc/hamiltonians/softabs_point.hpp]: methods are automatically inline

File: src/stan/mcmc/hmc/hamiltonians/diag_e_point.hpp [hunk 2]
```diff
@@ -53,6 +53,7 @@ namespace stan {
        *
        * @param writer Stan writer callback
        */
+      inline
```
Comment (bob-carpenter) [inline src/stan/mcmc/hmc/hamiltonians/diag_e_point.hpp]: We're trying to pack as much onto a single line as we can these days, but it's a minor issue, so I'm not going to hold this important request up for minor formatting.

**Comment by syclik** [comment]
Thanks @bgoodri. I'm in agreement with @bob-carpenter. Let's get this in and we'll get to the whitespace later.

**Comment by bgoodri** [comment]
Also, we still need a stand-alone test to catch when we break this again in the future.

**Pull request merged**